### PR TITLE
Add support for paused subscriptions

### PR DESCRIPTION
**Pull request by cartland**
https://developer.android.com/google/play/billing/subscriptions#pause

### CODE REVIEW

**Review by calren — APPROVED** [review]
nice!!

File: ClassyTaxiJava/app/src/main/java/com/sample/android/classytaxijava/ui/BillingViewModel.java [hunk 1]
```diff
@@ -96,14 +96,12 @@ public void openPlayStoreSubscriptions() {
     }
 
     /**
-     * Open account hold subscription.
+     * OpenOpen the subscription page on Google Play.
```
Comment (calren) [inline ClassyTaxiJava/app/src/main/java/com/sample/android/classytaxijava/ui/BillingViewModel.java:99]: small typo here

**Review by pyman2 — CHANGES_REQUESTED** [review]
Hi, Thanks for this, I'm just trying it out and I think we need to add this line to ClassyTaxiServer/src/play-billing/internal/purchases_impl.ts after line 156, as it is sending autoResumeTimeMillis to the app as a String:

    if (purchase.autoResumeTimeMillis) purchase.autoResumeTimeMillis = Number(purchase.autoResumeTimeMillis);

**Comment by cartland** [comment]
@pyman2 thank you! Good catch. I also updated the UI to show the date when the subscription will resume, so now it will actually use the autoResumeTimeMillis instead of just passing it to the app without using it.

**Comment by cartland** [comment]
@calren can you please take another look at the recent changes?

**Comment by calren** [comment]
nice, lgtm. and thanks @pyman2 for catching that!

**Comment by pyman2** [comment]
@cartland Now I have autoResumeTimeMillis thanks to your code, I'm hoping to use it to see whether a pause is still upcoming or has been cancelled by the user, but the values it sends aren't what I would expect.  I've asked Google about it here: 
 https://issuetracker.google.com/u/1/issues/179463140

**Comment by cartland** [comment]
@pyman2 thank you for submitted the information in the Issue Tracker! The engineering team will take a look at the details there.

**Review by tata406 — APPROVED** [review]

**Pull request merged**